recipes/mbedtls/all/conanfile.py

```diff
@@ -52,23 +53,25 @@ def configure(self):
             # The command line flags set are not supported on older versions of gcc
             raise ConanInvalidConfiguration("{}-{} is not supported by this recipe".format(self.settings.compiler, self.settings.compiler.version))
 
+        if tools.Version(self.version) >= "3.0.0":
```

Comment:
I don't really know how to handle this deprecation.
On other packages I've seen that we add a third value to the option ("deprecated") and make it the default.
In this case this would break the recipe compatibility with 2.x packages. I decided to leave it as is, force it to false on `Configure` and delete it from the `package_id`. It feels wrong, I'm open to suggestions.

Reply:
You can remove this option in `config_options()` if version is greater or equal than 3.0.0, then rely on `self.options.get_safe("zlib")`.
No deprecation here, we don't change the logic of the recipe, just follow a breaking change of upstream.

Reply:
Done